api/src/db/migrations/v16_add_user_preferences.ts

```diff
@@ -0,0 +1,113 @@
+import { v15UserData } from "./v15_add_retail_industry";
+
+export interface v16UserData {
+  user: v16BusinessUser;
+  onboardingData: v16OnboardingData;
+  formProgress: v16FormProgress;
+  taskProgress: Record<string, v16TaskProgress>;
+  licenseData: v16LicenseData | undefined;
+  preferences: v16Preferences;
+  version: number;
+}
+
+export const migrate_v15_to_v16 = (v15Data: v15UserData): v16UserData => {
+  return {
+    ...v15Data,
+    preferences: {
+      roadmapOpenSections: ["PLAN", "START"],
+      roadmapOpenSteps: [],
+    },
+    version: 16,
+  };
+};
+
+// ---------------- v16 types ----------------
+
+type v16TaskProgress = "NOT_STARTED" | "IN_PROGRESS" | "COMPLETED";
+type v16FormProgress = "UNSTARTED" | "COMPLETED";
+
+type v16BusinessUser = {
+  name?: string;
+  email: string;
+  id: string;
+  myNJUserKey?: string;
+};
+
+interface v16OnboardingData {
+  businessName: string;
+  industry: v16Industry | undefined;
+  legalStructure: v16LegalStructure | undefined;
+  municipality: v16Municipality | undefined;
+  liquorLicense: boolean;
+  homeBasedBusiness: boolean;
+  constructionRenovationPlan: boolean | undefined;
+}
+
+export type v16Municipality = {
+  name: string;
+  displayName: string;
+  county: string;
+  id: string;
+};
+
+type v16Industry =
+  | "restaurant"
+  | "e-commerce"
+  | "home-contractor"
+  | "cosmetology"
+  | "cleaning-aid"
+  | "retail"
+  | "generic";
+
+type v16LegalStructure =
+  | "sole-proprietorship"
+  | "general-partnership"
+  | "limited-partnership"
+  | "limited-liability-partnership"
+  | "limited-liability-company"
+  | "c-corporation";
+
+type v16NameAndAddress = {
+  name: string;
+  addressLine1: string;
+  addressLine2: string;
+  zipCode: string;
+};
+
+type v16LicenseData = {
+  nameAndAddress: v16NameAndAddress;
+  completedSearch: boolean;
+  lastCheckedStatus: string;
+  status: v16LicenseStatus;
+  items: v16LicenseStatusItem[];
+};
+
+type v16Preferences = {
+  roadmapOpenSections: v16SectionType[];
+  roadmapOpenSteps: number[];
```

Comment:
a nitpick, and no need to change this for this now but something to think about for the future. I know we'll need the roadmapOpenSteps eventually for an upcoming story, but since it's not used in this story, I would have waited to add this property of preferences until we actually need to use it, instead of having an empty array in there that we do nothing with until we get to that story later on

Reply:
my reasoning for this is twofold: 
- commits being a record of the work involved in each story. if we do a bit of work for another story in this one, then someone reading through this 5 months from now would be all "why are we adding this value if we don't use it??" because they aren't aware of the context of "oh we'll need it soon" so it feels out of place
- I know, and you know, that we are likely to do the openSteps story next. but we don't know. maybe our PMs will decide something else is more important. maybe it gets shuffled down the backlog for who-knows-how-long. maybe we never get around to it. and in the off-chance that we never get around to it, we'd have this extra field taking up space in our DB and in our brains for no reason. and we'd have to put work into cleaning it up! hence, preferring to keep it out until we actually need it, if it doesn't give us value now

Reply:
I think we added this property before we ended up agreeing to remove the acceptance criteria that had been added on. Not sure the best way to catch that after the fact, but something we may want to discuss (hopefully scope creep/revert isn't something that happens often)

Reply:
That makes total sense, and I don't want to suggest going back now and taking this out of the Preferences object, seems like more trouble than it's worth. More like something to keep in the back-of-mind regarding incremental changes for each story. But yeah I agree hopefully this is a unique case and most stories don't have as much shifting acceptance criteria anyways